### DISCUSSION TITLE
Any plans to support AirTags?

### DISCUSSION THREAD

**Discussion by ericblong** [mKeRix/room-assistant#666] [Ideas]
I guess it will be tricky if there's no possibility of running a companion app on the AirTag device to deal with the rolling random private address issue? But what if the companion app itself taps into Apple's Find My ecosystem which I think Apple has 'opened' uo to 3rd parties. Will that work?

**Comment by mKeRix** [comment]
I think the Find My network is limited to MFi licensees, so I'm not sure if we can get access there. In general I will take a look at whether we can support the AirTags somehow as well. I have one preordered for myself, so I'll use that for testing. :)

**Reply by Colorado4Wheeler** [reply]
You could always dig into the iCloud3 HACS plugin for HA, it's a super nice integration with iCloud for FindMy and might give you some ideas on integrating it.  I've got a handful of these on order for myself as well.

**Reply by Jens-Wymeersch** [reply]
Hello, any progression on this topic ?

**Reply by mKeRix** [reply]
Not yet unfortunately, due to lack of time. From my initial research on how AirTags and FindMy work it might be very tricky/not feasible to bring this to a room presence level tracking, since Apple secured it well (broadcasted address & public keys are both rotated, matching private key is stored securely on the iOS device).

I'm happy to accept contributions or pointers to posts of people that found a way of tracking these.

**Comment by mateuszdrab** [comment]
It would be nice if it was supported, I was thinking of using my dog's airtag to track his room presence and alert if he's somewhere where he shouldn't be. Might just get him a small ble beacon but can't find any that is small enough.

**Comment by TilBlechschmidt** [comment]
There might be another problem with this approach: If there is any Apple device close by the AirTag, then the AirTag will not send out any advertisements — meaning that unless you are not at home, AirTags won’t do anything and just sit there silently. However, it might be possible to get around this by not registering the AirTag with any Apple account. Though at that point it is just a very expensive iBeacon 🤷‍♂️

This article contains some very valuable information on this topic and is a generally interesting read: https://adamcatley.com/AirTag.html#key-facts-and-findings
